tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py

```diff
@@ -112,3 +116,68 @@ def test_streams_resilient_to_broker_down(self):
         self.assert_produce_consume("after_broker_stop")
 
         self.kafka.stop()
+
+    def test_streams_runs_with_broker_down_initially(self):
+        self.kafka.start()
+        node = self.kafka.leader(self.inputTopic)
+        self.kafka.stop_node(node)
+
+        broker_down_initially_in_seconds = 60
+
+        configs = self.get_configs(extra_configs=",application.id=starting_wo_broker_id")
+
+        # start streams with broker down initially
+        processor = StreamsBrokerDownResilienceService(self.test_context, self.kafka, configs)
+        processor.start()
+
+        processor_2 = StreamsBrokerDownResilienceService(self.test_context, self.kafka, configs)
+        processor_2.start()
+
+        processor_3 = StreamsBrokerDownResilienceService(self.test_context, self.kafka, configs)
+        processor_3.start()
+
+        time.sleep(broker_down_initially_in_seconds)
+
+        # now start broker
+        self.kafka.start_node(node)
+        # give broker time to start up
+        time.sleep(20)
+
+        # assert streams can process when starting with broker down
+        self.assert_produce_consume("running_with_broker_down_initially", num_messages=9)
+
+        self.kafka.stop()
```

Comment:
I guess we need to stop the 3 processors, too?

Reply:
I'm not sure why as I thought exiting the test method will clean everything up.